### PR TITLE
Decouple markdown validation in Markdig

### PR DESCRIPTION
**Pull request by 928PJY**
#2686

### CODE REVIEW

File: test/Microsoft.DocAsCode.MarkdigEngine.Tests/InclusionTest.cs [hunk 1]
```diff
@@ -535,9 +536,9 @@ public void TestFallback_InclusionWithCodeFences()
 			{
 				BasePath = "."
 			};
-			var service = new MarkdigMarkdownService(parameter);
-			//var rootMarked = service.Markup(Path.Combine(Directory.GetCurrentDirectory(), $"{uniqueFolderName}/root_folder"), root, fallbackFolders, "root.md");
-			var rootMarked = service.Markup("place", "holder");
+			var service = new MarkdigMarkdownServiceCreator().CreateMarkdigMarkdownService(parameter);
```
Comment (yufeih) [inline test/Microsoft.DocAsCode.MarkdigEngine.Tests/InclusionTest.cs]: Is it possible to NOT change the existing interface?

**Comment by OsmondJiang** [comment]
would you please add v3 tags so we can easily pick up PR only for V3?

File: src/docfx/SubCommands/DocumentBuilderWrapper.cs [hunk 2]
```diff
@@ -156,7 +156,7 @@ private static IEnumerable<Assembly> LoadPluginAssemblies(string pluginDirectory
                 typeof(TocDocumentProcessor).Assembly,
                 typeof(SchemaDrivenDocumentProcessor).Assembly,
                 typeof(UniversalReferenceDocumentProcessor).Assembly,
-                typeof(MarkdigServiceProvider).Assembly
+                typeof(MarkdigServiceProvider).Assembly,
```
Comment (qinezh) [inline src/docfx/SubCommands/DocumentBuilderWrapper.cs]: >, [](start = 55, length = 1)

why add `,` here?

Comment (928PJY) [reply src/docfx/SubCommands/DocumentBuilderWrapper.cs]: following stylecop `SA1413UseTrailingCommasInMultiLineInitializers`

---
In reply to: [186641787](https://github.com/dotnet/docfx/pull/2732#discussion_r186641787) [](ancestors = 186641787)

**Pull request merged**